pkg/engine/roleassignments.go

```diff
@@ -24,16 +24,13 @@ func createMSIRoleAssignment(identityRoleDefinition IdentityRoleDefinition) Role
 	return RoleAssignmentARM{
 		ARMResource: ARMResource{
 			APIVersion: "[variables('apiVersionAuthorizationUser')]",
-			DependsOn: []string{
```

Comment:
Because we are using the reference func below already, we don't need this DependsOn. See:

https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#implicit-dependency